RATIONALE.md

```diff
@@ -0,0 +1,13 @@
+# Notable rationale of wazero
+
+## wazeroir
+wazero's intermediate representation (IR) is called `wazeroir`. Compiling into an IR provides us a faster interpreter
+and a building block for a future JIT compilation engine. Both of these help answer demands for a more performant
+runtime vs interpreting Wasm directly (the `naivevm` interpreter).
+
+### Intermediate Representation (IR) design
+`wazeroir`'s initial design borrowed heavily from the defunct `microwasm` format (a.k.a. LightbeamIR). Notably,
+`wazeroir` doesn't have block operations: this simplifies the implementation.
```

Comment:
does this have any impact? For example, what do we lose without block operations? Is it just for humans reading the IR? or is there a functionality or perf degrade resulting?

Reply:
I think nothing to loose -- it is just a convenience layer for removing unused blocks in lightbeam code, but I realized that it's redundant, and we could achieve that without block operations (though I haven't done it yet).